Fix the name of the shared lib file and add the libversion to it.
diff --git a/IsoSpec++/CMakeLists.txt b/IsoSpec++/CMakeLists.txt
index badadb4..1e0785c 100644
--- a/IsoSpec++/CMakeLists.txt
+++ b/IsoSpec++/CMakeLists.txt
@@ -1,6 +1,12 @@
 project("IsoSpec")
 cmake_minimum_required(VERSION 2.6)
 
+## Version of the library (used in part for the SONAME)
+SET(ISOSPEC_VERSION_MAJOR "2")
+SET(ISOSPEC_VERSION_MINOR "0")
+SET(ISOSPEC_VERSION_PATCH "2")
+SET(ISOSPEC_VERSION "${ISOSPEC_VERSION_MAJOR}.${ISOSPEC_VERSION_MINOR}.${ISOSPEC_VERSION_PATCH}")
+
 set(my_sources
 cwrapper.cpp
 allocator.cpp
@@ -32,6 +38,15 @@ endif()
 
 add_library(IsoSpec SHARED ${my_sources})
 
+set_target_properties(IsoSpec
+	PROPERTIES OUTPUT_NAME IsoSpec++
+	CLEAN_DIRECT_OUTPUT 1
+	VERSION ${ISOSPEC_VERSION}
+	SOVERSION ${ISOSPEC_VERSION_MAJOR}
+	CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}"
+	)
+
+
 configure_file(IsoSpecConfig.cmake.in "${PROJECT_BINARY_DIR}/IsoSpecConfig.cmake" @ONLY)
 
 export(TARGETS IsoSpec FILE IsoSpecLibrary.cmake)
